src/Microsoft.ML.StandardLearners/FactorizationMachine/FactorizationMachineInterface.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Internal.CpuMath;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System.Runtime.InteropServices;
+
+using System.Security;
+
+namespace Microsoft.ML.Runtime.FactorizationMachine
+{
+    internal unsafe static class FieldAwareFactorizationMachineInterface
+    {
+        internal const string NativePath = "FactorizationMachineNative.dll";
+        public const int CbAlign = 16;
+
+        private static bool Compat(AlignedArray a)
+        {
+            Contracts.AssertValue(a);
+            Contracts.Assert(a.Size > 0);
+            return a.CbAlign == CbAlign;
+        }
+
+        private unsafe static float* Ptr(AlignedArray a, float* p)
+        {
+            Contracts.AssertValue(a);
+            float* q = p + a.GetBase((long)p);
+            Contracts.Assert(((long)q & (CbAlign - 1)) == 0);
+            return q;
+        }
+
+        [DllImport(NativePath), SuppressUnmanagedCodeSecurity]
+        public static extern void CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, int* /*const*/ fieldIndices, int* /*const*/ featureIndices,
+            float* /*const*/ featureValues, float* /*const*/ linearWeights, float* /*const*/ latentWeights, float* latentSum, float* response);
```

Comment:
>/*const*/ [](start = 51, length = 9)

are this comments here for a reason?

Reply:
@wschin just remove them, or need to comment on them?

---
In reply to: [196861706](https://github.com/dotnet/machinelearning/pull/383#discussion_r196861706) [](ancestors = 196861706)

Reply:
added this comment to the code. 

---
In reply to: [196948210](https://github.com/dotnet/machinelearning/pull/383#discussion_r196948210) [](ancestors = 196948210)